src/mainwindow.cpp

```diff
@@ -2551,7 +2552,7 @@ void MainWindow::closeEvent(QCloseEvent *event)
     if (!event->spontaneous() || !isVisible()) {
         return;
     }
-    if (m_hideToTray && m_trayIcon->isVisible()) {
+    if (m_hideToTray && m_trayIcon->isVisible() && QSystemTrayIcon::isSystemTrayAvailable()) {
```

Comment:
I suppose we could drop all the `m_trayIcon->isVisible()` checks, since it is always set to visible.